hostthedocs/util.py

```diff
@@ -2,17 +2,27 @@
 Provides utility methods.
 """
 
+import os
+from collections import namedtuple
+
 
 def get_filestream_from_request(request):
     """
     Extract the file-stream of the first file object from
     a :class:`werkzeug.wrappers.BaseRequest`.
 
     :param werkzeug.wrappers.BaseRequest request: the `werkzeug` request
-    :return: the file-stream of the first file within the request
+    :return: namedtuple the 'name', 'ext' (extension) and the 'stream' of the first file within the request
     :raises ValueError: if no files exist within the request
     """
     try:
-        return list(request.files.values())[0].stream
+        File = namedtuple('File', ['name', 'ext', 'stream'])
+        uplodaded_file = list(request.files.values())[0]
+        stream = uplodaded_file.stream
+        filename = uplodaded_file.filename
+        ext = os.path.splitext(filename)[1]
+        if ext.startswith('.'):
+            ext = ext[1:]
```

Comment:
The standard is usually to have extensions have leading dots. It looks like you're calling this 'extension' but using it as a 'format'. I would suggest renaming `fmt` in `unpack_project` to `ext` and handling the leading dot. That to me seems the most consistent.

You could rename this to `fmt` rather than `ext` and keep `unpack_project` how it is, but it introduces an ambiguous notion of a 'format' which we don't precisely define and is the extension anyway. So rather than do all that work for this app, I'd use an extension with a leading dot and change fmt to ext. If this were a library/these were used as public methods, I'd feel somewhat differently.